### PR TITLE
Json emitter memory fixes

### PR DESCRIPTION
**Pull request by gregschohn**
### Description
(from the commit) 
I've made the JsonEmitter implement AutoCloseable so that the underlying stream can be closed, which in turn will release the compositeByteBuf that it uses to buffer contents.
The NettyJsonBodySerializeHandler calls fireChannelRead(), then immediately called release on the same buffer that it just sent downstream on the channel. That latter release() has no been removed.
I've added a new unit test to help to test the NettyJsonBodySerializeHandler and to make sure that the memory accesses are solid. I've also updated the JsonEmitter test to not leak so that we're closer to a zero-leaks target.

* Category Bug fix
* Why these changes are required? stability & memory is finite
* What is the old behavior before changes and new behavior after changes? We leaked memory and opened ourselves up to corruption and now two spots for that fun to happen should be gone.

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-1313

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
Unit tests

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java [hunk 1]
```diff
@@ -34,16 +34,17 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
...
-    private void serializePayload(ChannelHandlerContext ctx, Map<String,Object> payload) throws IOException {
-        var pac = jsonEmitter.getChunkAndContinuations(payload, NUM_BYTES_TO_ACCUMULATE_BEFORE_FIRING);
-        while (true) {
-            ctx.fireChannelRead(new DefaultHttpContent(pac.partialSerializedContents));
-            pac.partialSerializedContents.release();
-            if (pac.nextSupplier == null) {
-                ctx.fireChannelRead(LastHttpContent.EMPTY_LAST_CONTENT);
-                break;
+    private void serializePayload(ChannelHandlerContext ctx, Map<String, Object> payload) throws IOException {
+        try (var jsonEmitter = new JsonEmitter(ctx.alloc())) {
```
Comment (okhasawn) [inline TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java:38]: Oh so this is why `JsonEmitter` implements `AutoCloseable`

**Review by okhasawn — APPROVED** [review]
Approved. Very similar to PR #317, should that one be closed?

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java [hunk 2]
```diff
@@ -34,16 +34,17 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         }
     }
 
-    private void serializePayload(ChannelHandlerContext ctx, Map<String,Object> payload) throws IOException {
-        var pac = jsonEmitter.getChunkAndContinuations(payload, NUM_BYTES_TO_ACCUMULATE_BEFORE_FIRING);
-        while (true) {
-            ctx.fireChannelRead(new DefaultHttpContent(pac.partialSerializedContents));
-            pac.partialSerializedContents.release();
```
[the remark this answers is not in the document]
Comment (gregschohn) [reply TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java:41]: fireChannelRead will call read, which will release.  This handler itself does not retain the buffer in its own capacity, so it shouldn't release the count either.

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java (hunk 1 above)
Comment (gregschohn) [reply TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java:38]: It implements AutoCloseable because it holds onto a CompositeByteBuf.  That needs to be released at some point and AutoCloseable makes that easy and self-documenting.  And this code here is where we take advantage of the auto-close trait.

**Comment by gregschohn** [comment]
See change [a884098](https://github.com/opensearch-project/opensearch-migrations/pull/325/commits/a8840982877b8f9b8d224cd1463c931b9bef1bbd) for cleanup items.
